src/webgpu/api/validation/getBindGroupLayout.spec.ts

```diff
@@ -0,0 +1,80 @@
+export const description = `
+  getBindGroupLayout validation tests.
+`;
+
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+
+import { ValidationTest } from './validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('index_range,explicit_layout')
+  .desc(
+    `
+  Test that a validation error is generated if the index exceeds the size of the bind group layouts
+  using a pipeline with an explicit layout.
+  `
+  )
+  .params(u => u.combine('index', [0, 1, 2, 3, 4, 5]))
+  .fn(async t => {
+    const { index } = t.params;
+
+    const maxBindGroups = t.device.limits.maxBindGroups;
+
+    const pipelineBufferBindings: GPUBindGroupLayoutEntry[] = [];
+    for (let binding = 0; binding < maxBindGroups; binding++) {
+      pipelineBufferBindings.push({
+        binding,
+        visibility: GPUShaderStage.COMPUTE,
+        buffer: { type: 'uniform' },
+      });
+    }
+
+    const pipelineBindGroupLayouts = t.device.createBindGroupLayout({
+      entries: pipelineBufferBindings,
+    });
+
+    const pipelineLayout = t.device.createPipelineLayout({
+      bindGroupLayouts: [pipelineBindGroupLayouts],
+    });
+
+    const pipeline = t.device.createRenderPipeline({
+      layout: pipelineLayout,
+      vertex: {
+        module: t.device.createShaderModule({
+          code: `
+            @vertex
+            fn main()-> @builtin(position) vec4<f32> {
+              return vec4<f32>(0.0, 0.0, 0.0, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+      },
+      fragment: {
+        module: t.device.createShaderModule({
+          code: `
+            @fragment
+            fn main() -> @location(0) vec4<f32> {
+              return vec4<f32>(0.0, 1.0, 0.0, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+        targets: [{ format: 'rgba8unorm' }],
+      },
+    });
+
+    const shouldError = index >= maxBindGroups;
```

Comment:
With #3506, this should error if index >= 1, since 1 is the number of bind group layouts in `pipelineLayout`.

Reply:
yes, `pipelineLayout` only has `pipelineBindGroupLayouts` as for `bindGroupLayouts`. Thanks for correcting me!